packages/jira-adapter/src/filters/request_type.ts

```diff
@@ -0,0 +1,180 @@
+import { ISSUE_VIEW_TYPE, REQUEST_FORM_TYPE, REQUEST_TYPE_NAME } from '../constants'
+import { deployChanges, defaultDeployChange } from '../deployment/standard_deployment'
+import JiraClient from '../client/client'
+import { layoutConfigItem } from './layouts/layout_types'
+import { LayoutTypeName, getLayoutResponse, isIssueLayoutResponse, layoutTypeNameToDetails } from './layouts/layout_service_operations'
+
+const { isDefined } = lowerDashValues
+
+type statusType = {
+  id: ReferenceExpression
+  original: string
+  custom: boolean
+}
+
+const fieldsToIgnore = [
+  'issueView',
+  'requestForm',
+  'workflowStatuses',
+  'avatarId',
+  'groupIds',
+]
+
+
+const deployWorkflowStatuses = async (
+  change: ModificationChange<InstanceElement> | AdditionChange<InstanceElement>,
+  client: JiraClient): Promise<void> => {
+  if ((isAdditionChange(change))
+    || (isModificationChange(change)
+    && !isEqualValues(change.data.before.value.workflowStatuses, change.data.after.value.workflowStatuses))) {
+    const instance = getChangeData(change)
+    const parent = getParent(instance)
+    const workflowStatuses = instance.value.workflowStatuses.map((status: statusType) => ({
+      id: status.id?.value.value.id,
+      original: status.id?.value.value.name,
+      custom: status.custom,
+    }))
+    await client.post({
+      url: `/rest/servicedesk/1/servicedesk-data/${parent.value.key}/request-type/${instance.value.id}/workflow`,
+      data: workflowStatuses,
+    })
+  }
+}
+
+const isRequesTypeDetailsChange = (
+  change: ModificationChange<InstanceElement>,
+  fieldName: string,
+): boolean => fieldName === 'requestForm'
+  && (change.data.before.value.name !== change.data.after.value.name
+    || change.data.before.value.description !== change.data.after.value.description
+    || change.data.before.value.helpText !== change.data.after.value.helpText)
+
+
+const deployRequestTypeLayout = async (
+  change: Change<InstanceElement>,
+  client: JiraClient,
+  typeName: LayoutTypeName,
+): Promise<void> => {
+  const requestType = getChangeData(change)
+  const { fieldName } = layoutTypeNameToDetails[typeName]
+  if (fieldName === undefined) {
+    return undefined
+  }
+  if ((isAdditionChange(change))
+  || (isModificationChange(change)
+  && (!isEqualValues(change.data.before.value[fieldName], change.data.after.value[fieldName])
+  || isRequesTypeDetailsChange(change, fieldName)))) {
+    const layout = requestType.value[fieldName]
+    const items = layout.issueLayoutConfig.items.map((item: layoutConfigItem) => {
+      if (isResolvedReferenceExpression(item.key)) {
+        const key = item.key.value.value.id
+        return {
+          type: item.type,
+          sectionType: item.sectionType.toLocaleLowerCase(),
+          key,
+          data: {
+            name: item.key.value.value.name,
+            type: item.key.value.value.type ?? item.key.value.value.schema.system,
+          },
+        }
+      }
+      return undefined
```

Comment:
1. nit: better to reverse the conditions for clarity + less nested code
```suggestion
      if (!isResolvedReferenceExpression(item.key)) {
        return undefined
      }
      const key = item.key.value.value.id
      return {
        type: item.type,
        sectionType: item.sectionType.toLocaleLowerCase(),
        key,
        data: {
          name: item.key.value.value.name,
          type: item.key.value.value.type ?? item.key.value.value.schema.system,
        },
      }
```
2. when do we expect the reference to not be resolved? if it's an error flow let's add an error log (or warning/info/debug if this is a valid scenario)
3. is there a valid scenario where some items are valid and some are not? is there any risk in deploying only some of the items the user requested?

Reply:
Its not suppose to happen. It just in case. I will add a error log